### PR TITLE
ROX-12750: Data migration for permission grouping / deprecation.

### PR DESCRIPTION
**Pull request by rhybrillou**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

**Comment by rhybrillou** [comment]
Current dependencies on/for this PR:
* master
  * **PR #5075** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/5075" target="_blank">Graphite</a> 
    * **PR #5091** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/5091" target="_blank">Graphite</a> 
      * **PR #5092** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/5092" target="_blank">Graphite</a> 
        * **PR #5093** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/5093" target="_blank">Graphite</a> 
          * **PR #5094** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/5094" target="_blank">Graphite</a> 
            * **PR #5095** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/5095" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/5095?utm_source=stack-comment).

File: migrator/migrations/m_174_to_m_175_group_permissions/migration.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+		newPermissionSet.ResourceToAccess = make(map[string]storage.Access, len(obj.GetResourceToAccess()))
+		for resource, accessLevel := range obj.GetResourceToAccess() {
+			if replacement, found := replacements[resource]; found {
+				changed = true
+				resource = replacement
+			}
+			newPermissionSet.ResourceToAccess[resource] =
+				propagateAccessForPermission(resource, accessLevel, newPermissionSet.GetResourceToAccess())
+		}
+		if changed {
```
Comment (janisz) [inline migrator/migrations/m_174_to_m_175_group_permissions/migration.go:100]: nit: flip if to keep less indents

File: migrator/migrations/m_174_to_m_175_group_permissions/permissionsetpostgresstore/postgres_plugin.go [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+// New returns a new Store instance using the provided sql instance.
+func New(db *postgres.DB) Store {
+	return &storeImpl{
+		db: db,
+	}
+}
+
+//// Helper functions
+
+func insertIntoPermissionSets(ctx context.Context, batch *pgx.Batch, obj *storage.PermissionSet) error {
```
Comment (janisz) [inline migrator/migrations/m_174_to_m_175_group_permissions/permissionsetpostgresstore/postgres_plugin.go:67]: unused `ctx`

File: migrator/migrations/m_174_to_m_175_group_permissions/permissionsetpostgresstore/postgres_plugin.go [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+// This file is a copy of central/role/store/permissionset/postgres/store.go at the time of the migration.
+// The content was trimmed down to the necessary functions (UpsertMany and Walk), and to the requirements
+// of migration store files (reference migrations frozen schema and removed metrics collection as well as
+// scoped access control)
+
+const (
+	baseTable = "permission_sets"
```
Comment (janisz) [inline migrator/migrations/m_174_to_m_175_group_permissions/permissionsetpostgresstore/postgres_plugin.go:27]: unused

File: migrator/migrations/m_174_to_m_175_group_permissions/permissionsetpostgresstore/postgres_plugin.go [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+
+			// clear the input rows for the next batch
+			inputRows = inputRows[:0]
+		}
+	}
+
+	return err
+}
+
+func (s *storeImpl) acquireConn(ctx context.Context, op ops.Op, typ string) (*postgres.Conn, func(), error) {
```
Comment (janisz) [inline migrator/migrations/m_174_to_m_175_group_permissions/permissionsetpostgresstore/postgres_plugin.go:154]: unused `op, typ`

**Review by janisz — APPROVED** [review]
some nitpicks

File: migrator/migrations/m_174_to_m_175_group_permissions/migration.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+		SensorUpgradeConfig: Administration,
+		ServiceIdentity:     Administration,
+	}
+)
+
+func init() {
+	migrations.MustRegisterMigration(migration)
+}
+
+// Additional code to support the migration
```
Comment (dhaus67) [inline migrator/migrations/m_174_to_m_175_group_permissions/migration.go:67]:
```suggestion

```

**Review by dhaus67 — APPROVED** [review]

File: pkg/migrations/internal/seq_num.go [hunk 6]
```diff
@@ -4,7 +4,7 @@ var (
 	// CurrentDBVersionSeqNum is the current DB version number.
 	// This must be incremented every time we write a migration.
 	// It is a shared constant between central and the migrator binary.
-	CurrentDBVersionSeqNum = 172
+	CurrentDBVersionSeqNum = 175
```
Comment (dashrews78) [inline pkg/migrations/internal/seq_num.go:7]: I assume you have a couple of migrations stacked up in review. just make sure they are pushed in order or update the sequence accordingly.

**Review by dashrews78 — APPROVED** [review]
LGTM, just be cautious of the migrations you have in flight so sequence is correct when pushed.

**Pull request merged**